server/src/main/java/org/opensearch/index/shard/CloseableRetryableRefreshListener.java

```diff
@@ -71,7 +68,12 @@ protected TimeValue getNextRetryInterval() {
         return null;
     }
 
-    private void scheduleRetry(TimeValue interval, String retryThreadPoolName, boolean didRefresh, boolean isRetry) {
+    private void scheduleRetry(TimeValue interval, String retryThreadPoolName, boolean didRefresh) {
+        // If the underlying listener has closed, then we do not allow even the retry to be scheduled
+        if (closed.get()) {
+            return;
+        }
```

Comment:
Do we need to add a UT for this scenario?

Reply:
There are couple of tests that have been added. I will recheck and add more tests for couple of other combinations.